src/librustdoc/clean/inline.rs

```diff
@@ -477,6 +477,7 @@ fn build_module(
                                 }],
                             },
                             did: None,
+                            attrs: None,
```

Comment:
@jyn514 Because if I understand correctly, I would have to do

```
let attrs = Box::new(cx.tcx().get_attrs(def_id).clean(cx));
```

in `print_item.rs`, and it didn't seem appropriate to call `clean()` methods at that stage, not least because `clean()` expects a `DocContext`, and we don't have that in `print_item.rs`.

Let me know if you see a better way!

Reply:
> it didn't seem appropriate to call clean() methods at that stage, not least because clean() expects a DocContext, and we don't have that in print_item.rs.

Got it, thanks. At this point, I think I would extract a free function in `clean::types` for `stability_class` so you only have to pass in the DefId of the item, not a whole `clean::Item`. Note that `stability_class` never uses the attributes, only the DefId. I would rather not make rustdoc slower just because the API is bad.

You'll have to also extract free functions for `stability`, `deprecation`, and `is_fake`; all that seems fine.

cc https://github.com/rust-lang/rust/issues/83183

Reply:
I could do that, but I would still have to pass in `attrs`, and it also wouldn't solve the issue with the fake `Item`. You see, the problem isn't so much with `stability_class`, but rather with `extra_info_tags`. It really does seem to need an `Item` with `attrs` on it.

Reply:
Hmm, ok. I guess this is a larger refactor than really belongs in this PR, I opened https://github.com/rust-lang/rust/issues/84304.

Reply:
Aha, there's a simpler way: `impl Clean for [ast::Attribute]` doesn't use DocContext at all: https://github.com/rust-lang/rust/blob/b021beeb6021d8885dc3f21e75a7f56502849633/src/librustdoc/clean/mod.rs#L242
So you can remove the `attrs` field and call `let attrs = Box::new(Attributes::from_ast(tcx.sess().diagnostic(), tcx.get_attrs(def_id), None));` on-demand instead.